renku/domain_model/dataset.py

```diff
@@ -380,6 +381,7 @@ def __init__(
         self.title: Optional[str] = title
         self.version: Optional[str] = version
         self.annotations: List["Annotation"] = annotations or []
+        self._datadir: Optional[str] = str(datadir) if datadir else None
```

Comment:
I suggest that we use the same pattern for new attributes that we add to a Persistent model. Before, we've added the attribute to the class and then set it for individual instances if it had a value. We should do the same here. Also, we can call it `datadir` instead of `_datadir`.

Reply:
yes I had it as a property first, then this made sense. I'll change it. :+1: